client/components/detail.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+export default class Detail extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      post: []
+    };
+  }
+
+  componentDidMount() {
+    const targetId = window.location.hash.split('/');
+    fetch(`/api/post/${targetId[1]}`)
+      .then(res => res.json())
```

Comment:
Instead of string manipulation from the hash, use a prop here instead.
```js
componentDidMount() {
  fetch(`/api/post/${this.props.postId}`)
  // ...
}
```